test/IntegrationTests/MSTest.Acceptance.IntegrationTests/TestRunParametersTests.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Testing.Platform.Acceptance.IntegrationTests;
+using Microsoft.Testing.Platform.Acceptance.IntegrationTests.Helpers;
+
+namespace MSTest.Acceptance.IntegrationTests;
+
+[TestGroup]
+public sealed class TestRunParametersTests : AcceptanceTestBase
+{
+    private readonly TestAssetFixture _testAssetFixture;
+
+    public TestRunParametersTests(ITestExecutionContext testExecutionContext, TestAssetFixture testAssetFixture)
+        : base(testExecutionContext)
+    {
+        _testAssetFixture = testAssetFixture;
+    }
+
+    [ArgumentsProvider(nameof(TargetFrameworks.All), typeof(TargetFrameworks))]
+    public async Task CanProvideTestRunParameters(string tfm)
+    {
+        var testHost = TestHost.LocateFrom(_testAssetFixture.ProjectPath, TestAssetFixture.ProjectName, tfm);
+        TestHostResult testHostResult = await testHost.ExecuteAsync("--settings my.runsettings --test-parameter MyParameter2=MyValue2 MyParameter3=MyValue3");
```

Comment:
I am talking about this line, I thought that parameters don't consume arrays, and so the `MyParameter3=MyValue3` should trigger an error, even though the parameter is oneOrMany, because it requires `--test-parameter MyParameter2=MyValue2 --test-parameter MyParameter3=MyValue3` syntax.

Maybe I just remember it wrong?

Reply:
Both syntaxes are supported (see https://learn.microsoft.com/dotnet/standard/commandline/syntax#argument-arity). I'll add a variant with the other syntax so we have both documented.